Code/Tools/ProjectManager/Source/GemCatalog/GemInfo.cpp

```diff
@@ -100,4 +100,56 @@ namespace O3DE::ProjectManager
     {
         return (m_displayName < gemInfo.m_displayName);
     }
+
+    GemInfo::Platforms GemInfo::GetPlatformFromString(const QString& platformText)
+    {
+        if(platformText == "Windows")
+        {
+            return GemInfo::Platform::Windows;
+        }
+        else if(platformText == "Linux")
+        {
+            return GemInfo::Platform::Linux;
+        }
+        else if(platformText == "Android")
+        {
+            return GemInfo::Platform::Android;
+        }
+        else if(platformText == "iOS")
+        {
+            return GemInfo::Platform::iOS;
+        }
+        else if(platformText == "macOS")
+        {
+            return GemInfo::Platform::macOS;
+        }
```

Comment:
This shouldn't be hardcoded.

Instead the existing set of Platform "defines" defined in [PlatformDefaults.h](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/PlatformId/PlatformDefaults.h) should be used.

The [PlatformHelper::GetPlatformName](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/PlatformId/PlatformDefaults.h#L109-L110) function can be used to translated an enum to a string

Reply:
I definitely agree in principle, but I'm noticing there's a mismatch between the `Windows` value in the ProjectManager codebase as compared to the `PlatformDefaults` values of `AZ::PC` and `PlatformId::PLATFORM_WINDOWS_64 --> Win64`.

As such I'm not sure I can take action on this suggestion in the scope of this PR. I'd need to come back to this one and revise, because there are more places in the Project Manager where these OS values are hardcoded in this manner.

Reply:
however I won't be able to use the codename for Mac `PlatformCodeNameMac`, because the string value is `Mac`, whereas in Project Manager it's `macOS`.

Reply:
We should be consistent in our naming.  Let's find out if there's any reason why we cannot use the AzCore names or at least the code names.

Reply:
Well "macOS" kind of break this situation of using the Codename.
However the term "macOS" should be capitalized to "MacOS".

Reply:
Sounds good. So would it be fine if I note this in a github issue and address it after this PR?

Reply:
OK I'm fine with deferring this for a later PR